### PR TITLE
feat: [FFM-8354]: Serve Percentage Rollout sub-section

### PR DESCRIPTION
**Pull request by knagurski**
This PR adds the Serve Percentage Rollout to Target Group sub-section.

Screenshot 2023-11-07 at 00 27 55

### CODE REVIEW

File: src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/__tests__/flagChangesValidationSchema.test.ts [hunk 1]
```diff
@@ -11,18 +11,19 @@ import { CFPipelineInstructionType } from '../../types'
 describe('flagChangesValidationSchema', () => {
   const getStringMock = jest.fn().mockImplementation(str => str)
 
-  test('it should throw with 5 errors when each type is specified but not valid', async () => {
+  test('it should throw with 7 errors when each type is specified but not valid', async () => {
```
Comment (crystal-fong) [inline src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/__tests__/flagChangesValidationSchema.test.ts:14]: 5 instructions types here but 7 errors?

File: src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/subSections/ServePercentageRolloutToTargetGroup/ServePercentageRolloutToTargetGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            return getPercentageRolloutVariationsArrayTest(getString)
+          } else {
+            return Yup.string().required(
+              getString('cf.featureFlags.flagPipeline.validation.servePercentageRollout.variations')
+            )
+          }
+        })
+      })
+    })
+  })
```
Comment (laura-mannus) [inline src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/subSections/ServePercentageRolloutToTargetGroup/ServePercentageRolloutToTargetGroup.tsx:56]: great job, this yup schema gives me heart palpitations

Comment (knagurski) [reply src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/subSections/ServePercentageRolloutToTargetGroup/ServePercentageRolloutToTargetGroup.tsx:56]: Would you believe that this is the simplified version? 😅

File: src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/__tests__/flagChangesValidationSchema.test.ts (hunk 1 above)
Comment (knagurski) [reply src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/__tests__/flagChangesValidationSchema.test.ts:14]: Yeah, confusing, but a couple of the instructions have multiple fields that are independently validated.

**Review by laura-mannus — APPROVED** [review]

**Comment by laura-mannus** [comment]
Not confident enough in how to play around with the functionality but you're gonna cover it in the next sync so I am pleased to approve :)

File: src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/__tests__/flagChangesValidationSchema.test.ts (hunk 1 above)
Comment (crystal-fong) [reply src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/FlagChangesV2/__tests__/flagChangesValidationSchema.test.ts:14]: gotcha 👍

**Review by crystal-fong — APPROVED** [review]

**Review by Kira-Jennings — APPROVED** [review]

**Pull request merged**